hamp_radar/serde_flight.py

```diff
@@ -0,0 +1,160 @@
+from collections.abc import Iterable
+from typing import List
+from pathlib import Path
+
+from geometries import (
+    MultiMainBlockGeometry,
+    PdsFileGeometry,
+    DatasetBlockGeometry,
+    DatasetGeometry,
+    FlightGeometry,
+)
+from serde_pds_files import (
+    get_pdsfile_geometries,
+    write_serialized_geometry,
+)
+
+
+def ppar_for_dataset(
+    filename: Path, mmbg: MultiMainBlockGeometry
+) -> DatasetBlockGeometry:
+    """
+    Creates a DatasetBlockGeometry object with only a single PPAR from a MultiMainBlockGeometry
+    instance. Assumes mmbg.tag == "PPAR".
+
+    If mmbg.count > 1 a new mmbg is created containing only the last PPAR in the mmbg.
+
+    Args:
+        filename (Path): The path to the .pds file associated with the mmbg.
+        mmbg (MultiMainBlockGeometry): The multi-main block geometry object.
+
+    Returns:
+        DatasetBlockGeometry: A new dataset block geometry object with only the latest PPAR in the mmbg.
+
+    Raises:
+        ValueError: If there are two consecutive PPAR subblocks, indicating corrupt data.
+    """
+    if len(mmbg.subblocks) > 1:
+        raise ValueError(
+            "Two consecutive PPAR subblocks should never occuur, corrupt data?"
+        )
+
+    if mmbg.count > 1:
+        # only use last PPAR for DSP configuration
+        offset = mmbg.offset + mmbg.step * (mmbg.count - 1)
+        count = 1
+        mmbg = MultiMainBlockGeometry(
+            mmbg.tag, offset, count, mmbg.step, mmbg.subblocks
+        )
+
+    ppar = DatasetBlockGeometry(filename, mmbg)
+    return ppar
+
+
+def convert_to_datasetgeometries(
+    pfgs: Iterable[PdsFileGeometry],
+) -> List[DatasetGeometry]:
+    """
+    Converts an iterable of PdsFileGeometry instances into a list of DatasetGeometry instances.
+
+    This function converys a collection of PdsFileGeometry instances into
+    DatasetGeometry instances. It identifies "PPAR" tags to start new datasets and appends
+    data blocks to the most recent dataset. If consecutibe PPAR tags are found withou
+    data in between, the ppar of the current dataset is updated to match the last given PPAR.
+
+    Args:
+        pfgs (Iterable[PdsFileGeometry]): An iterable of PdsFileGeometry instances, e.g. for a flight.
+
+    Returns:
+        List[DatasetGeometry]: A list of DatasetGeometry instances, e.g. for each dataset in a flight.
+
+    Raises:
+        UserWarning: If data is found before the first "PPAR" tag in the PdsFileGeometry's mainblocks.
+    """
+    import warnings
+
+    datasets, is_ppar = [], False
```

Comment:
The two aren't really related, so initializing them together probably confuses more than it helps.
```suggestion
    datasets = []
    is_ppar = False
```